### DISCUSSION TITLE
DietPi-Software | ownCloud: Migrate to v2.0 Go "Infinite Scale" version

### DISCUSSION THREAD

**Discussion by MichaIng** [MichaIng/DietPi#6210] [Software Request] [answered]
https://owncloud.com/infinite-scale/

**Comment by Joulinar** [comment]
do you know what the difference between OCIS and ownCloud 10 is?

**Reply by MichaIng** [reply]
Written in Go instead of PHP, hence doesn't use a dedicate webserver etc but is an all integrated binary. Basically a complete rework. I thought they would support both versions for a longer while, but the classic ownCloud is not updated anymore.

**Reply by Joulinar** [reply]
Hmm that might be a challenge to avoid port conflict with already installed web server stack.

**Reply by MichaIng** [reply]
I'd install it on a different port, like we do with all other software titles with embedded webserver. I haven't checked yet, but I think it can be manually changed to run on port 80/443, and otherwise a reverse proxy or port switch in forwarding can be used.

**Comment by Joulinar** [comment]
> otherwise a reverse proxy or port switch in forwarding can be used

In this case we might need to provide some example configs.

**Reply by MichaIng** [reply]
I do not see a reason why we necessarily do more efforts for ownCloud in this regards than we do in all other cases, Syncthing, etc. But of course, it would be generally nice to have some proxy examples for web applications.

I've btw no idea whether there is a great migration path for users from ownCloud 10 to ownCloud v2.0/Infinite Scale. But I guess so. Alternatively there is a migration path to Nextcloud, which means that the surrounding setup stays the same, with webserver and PHP.

**Comment by Inkapa** [comment]
Are there any updates around this? 
OCIS version 3.0 released 5 days ago and I've been interested in getting it working on my rather weak PI 3A+, which is how I've stumbled upon DietPi. But I'm not sure I have the knowledge and patience to configure it without an example config or some documentation 😅

**Reply by MichaIng** [reply]
No one started working on it yet. I would also need to start from zero, reading into it. Best documentation page for a full setup bare metal with systemd seems to be this one: https://doc.owncloud.com/ocis/next/depl-examples/bare-metal.html
1. Download binary
2. Create environment file with basic path and network information
3. Initialise a config file
4. Create a service and run user to start OCIS

What can be seen a little complicated is that every feature ("service") of OCIS runs as own process and listens on an own port. The `PROXY_URL` in the environment file indeed is an (internal) proxy to allow access to all individual services on a single port. I guess the YAML config file will maintain a mapping for this proxy.

**Comment by samolego** [comment]
Any news on this?

**Reply by Joulinar** [reply]
Nothing has been done on this yet

**Comment by MichaIng** [comment, marked as answer]
PR open: #7886

**Reply by samolego** [reply]
If interested: OwnCloud Infin. Scale was again forked (like with nextcloud), with some devs moving to the new fork, OpenCloud: https://www.heise.de/en/news/Ex-ownCloud-devs-seek-new-start-at-OpenCloud-Owncloud-owner-wants-to-sue-10254438.html

**Reply by MichaIng** [reply]
That is a different request already: #7483

**Reply by samolego** [reply]
Oh sorry, should've searched first.

Thanks for OCIS impl though :)!
